content/en/docs/instrumentation/erlang/testing.md

```diff
@@ -0,0 +1,132 @@
+---
+title: "Testing"
+weight: 40
+---
+
+When relying on OpenTelemetry for your Observability needs, it can be important to test that certain spans are created and attributes correctly set. For example, can you be sure that you attaching the right metadata to data that ultimately powers an SLO? This document covers an approach to that kind of validation.
+
+## Setup
+Only the `opentelemetry` and `opentelemetry_api` libraries are required for testing in Elixir/Erlang:
+
+{{< tabs Erlang Elixir >}}
+
+{{< tab >}}
+{deps, [{opentelemetry_api, "~> 1.0"},
+        {opentelemetry, "~> 1.0"}]}.
+{{< /tab >}}
+
+{{< tab >}}
+def deps do
+  [
+    {:opentelemetry_api, "~> 1.0"},
+    {:opentelemetry, "~> 1.0"}
+  ]
+end
+{{< /tab >}}
+
+{{< /tabs >}}
+
+Set your `exporter` to `:none` and the span processor to `:otel_simple_processor`. This ensure that your tests don't actually export data to a destination, and that spans can be analyzed after they are processed.
+
+{{< tabs Erlang Elixir >}}
+
+{{< tab >}}
+%% config/sys.config.src
+{opentelemetry,
+  [{traces_exporter, none},
+   {processors,
+     [{otel_simple_processor, #{}}]}]}
+{{< /tab >}}
+
+{{< tab >}}
+# config/test.exs
+import Config
+
+config :opentelemetry,
+    traces_exporter: :none
+
+config :opentelemetry, :processors, [
+  {:otel_simple_processor, %{}}
+]
+{{< /tab >}}
+
+{{< /tabs >}}
+
+A modified version of the `hello` function from the [Getting Started](/docs/instrumentation/erlang/getting-started/) guide will serve as our test case:
+
+{{< tabs Erlang Elixir >}}
+
+{{< tab >}}
+%% apps/otel_getting_started/src/otel_getting_started.erl
+-module(otel_getting_started).
+
+-export([hello/0]).
+
+-include_lib("opentelemetry_api/include/otel_tracer.hrl").
+
+hello() ->
+    %% start an active span and run a local function
+    ?with_span(<<"operation">>, #{}, fun nice_operation/1).
+
+nice_operation(_SpanCtx) ->
+    ?set_attributes([{a_key, <<"a value">>}]),
+    world
+{{< /tab >}}
+
+{{< tab >}}
+# lib/otel_getting_started.ex
+defmodule OtelGettingStarted do
+  require OpenTelemetry.Tracer, as: Tracer
+
+  def hello do
+    Tracer.with_span "operation" do
+      Tracer.set_attributes([{:a_key, "a value"}])
+      :world
+    end
+  end
+end
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Testing
+
+{{< tabs Erlang Elixir >}}
+
+{{< tab >}}
+TODO
```

Comment:
```suggestion
-module(otel_getting_started_SUITE).

-compile(export_all).

-include_lib("stdlib/include/assert.hrl").
-include_lib("common_test/include/ct.hrl").

-include_lib("opentelemetry/include/otel_span.hrl").

-define(assertReceive(SpanName),
        receive
            {span, Span=#span{name=SpanName}} ->
                Span
        after
            1000 ->
                ct:fail("Did not receive the span after 1s")
        end).

all() ->
    [greets_the_world].

init_per_suite(Config) ->
    application:load(opentelemetry),
    application:set_env(opentelemetry, processors, [{otel_simple_processor, #{}}]),
    {ok, _} = application:ensure_all_started(opentelemetry),
    Config.

end_per_suite(_Config) ->
    _ = application:stop(opentelemetry),
    _ = application:unload(opentelemetry),
    ok.

init_per_testcase(greets_the_world, Config) ->
    otel_simple_processor:set_exporter(otel_exporter_pid, self()),
    Config.

end_per_testcase(greets_the_world, _Config) ->
    otel_simple_processor:set_exporter(none),
    ok.

greets_the_world(_Config) ->
    %% SpanName = <<"operation">>,
    otel_getting_started:hello(),

    ExpectedAttributes = otel_attributes:new(#{a_key => <<"a_value">>}, 128, infinity),
    #span{attributes=ReceivedAttributes} = ?assertReceive(<<"operation">>),

    %% use an assertMatch instead of matching in the `receive'
    %% so we get a nice error message if it fails
    ?assertMatch(ReceivedAttributes, ExpectedAttributes),

    ok.
```